kanidmd/src/lib/macros.rs

```diff
@@ -240,41 +240,77 @@ macro_rules! run_create_test {
         use crate::prelude::*;
         use crate::schema::Schema;
         use crate::utils::duration_from_epoch_now;
-
-        let mut au = AuditScope::new("run_create_test", uuid::Uuid::new_v4(), None);
-        lperf_segment!(&mut au, "plugins::macros::run_create_test", || {
-            let qs = setup_test!(&mut au, $preload_entries);
-
-            let ce = match $internal {
-                None => CreateEvent::new_internal($create_entries.clone()),
-                Some(e_str) => unsafe {
-                    CreateEvent::new_impersonate_entry_ser(e_str, $create_entries.clone())
-                },
-            };
-
-            {
-                let qs_write = qs.write(duration_from_epoch_now());
-                let r = qs_write.create(&mut au, &ce);
-                debug!("test result: {:?}", r);
-                assert!(r == $expect);
-                $check(&mut au, &qs_write);
-                match r {
-                    Ok(_) => {
-                        qs_write.commit(&mut au).expect("commit failure!");
+        use crate::logbuffer::LogBuffer;
+
+        use tracing::{self, Level};
+        use tracing_subscriber::FmtSubscriber;
+        use tracing_subscriber::fmt::format::{FmtSpan, debug_fn};
+        use tracing_subscriber::field::MakeExt;
+
+        // https://docs.rs/tracing-subscriber/0.2.18/tracing_subscriber/fmt/struct.SubscriberBuilder.html
+        let subscriber = FmtSubscriber::builder()
+            // SETTINGS
+            .with_max_level(Level::TRACE)
+            .with_ansi(false) // no colors (good for .log files)
+
+            // JSON
```

Comment:
We'll need some way to pass this configuration in in the future.

Reply:
Where are we accessing the config in functions? Is it eventually going to be an object that's passed around? Or a global var? I agree that this should be configurable though.

Reply:
It's not so simple. There are some values that are global, and some that are "per request" so each subscriber may be uniquely configured - or could be re-configured during a request depending on which stage we are in of the process. 

For now we'll just assume some global defaults though, let's not overthink it yet. But we'll need a way to say "show me the current subscriber" for an event.

The main barrier right now will be the batching requirement. I think it may be time to make a new "side project" and experiment there in a small scale rather than here. We sohuld probably have a design doc and a talk about it?